src/auth/auth.service.ts

```diff
@@ -40,6 +33,22 @@ export class AuthService {
     return userRecordWithoutPassword;
   }
 
+  private signUser(id: string, role: RoleType) {
+    return this.jwtService.sign({
+      sub: id,
+      claim: role,
+    });
+  }
+
+  public googleLogin(req: any) {
```

Comment:
Use same name as in the controller `loginWithGoogleProvider`

```suggestion
  public loginWithGoogleProvider(providerUserProfile: Profile) {
```

Reply:
Will do